src/app/content/highlights/components/utils/useCallbackFor.ts

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+// Some functions get recreated but shouldn't be reacted to
+// This makes a non-changing callback
+export default function useCallbackFor<T extends Array<unknown>>(fn: (...args: T) => unknown) {
+    const ref = React.useRef(fn);
+    const cb = React.useCallback(
```

Comment:
unless i'm missing something, the presence of this useCallback hook isn't really doing anything. since you're passing `[]` for the dependencies, you might as well define this as 

```
export default function useInitialValue<T>(thing: T) {
  return React.useRef(thing).current;
}
```

Reply:
Using an empty dependency list means define it once, not redefine it every time.

Reply:
yes, `return React.useRef(thing).current;` also only uses the first thing passed in and ignores updates. to update it every time you'd have to separately set `ref.current = thing` in each call. i think my suggestion in the earlier comment is functionally the same as yours but could be used in more ways because its not limited to functions

Reply:
If I return `ref.current` each time, it's still changing. I need a wrapper that doesn't change, and I really only needed it for function calls. I didn't want to change from calling a function to calling an object method or something.

Reply:
it does not change each time https://codesandbox.io/s/naughty-ellis-txftc6?file=/src/App.js

theres something up with the code sandbox it double renders initially and that causes the counter to start at two, but you can click the button and see that its rendering and not continuing to affect the ref or effect

Reply:
It looks like it's not needed after all. There's a quirk in lint that doesn't want `props.function` as a dependency and insists on `props` instead, but if I just assign the function to a variable, the dependency is ok. Everything works without being wrapped in `useCallbackFor`, so I've removed it.